benchmark/serializer.py

```diff
@@ -48,14 +49,22 @@ def serialize(dataset_name, dataset_path, serialized_graph_path):
 
     for s in serialize_queries:
         logging.info('Executing query: %s', s)
-        try:
-            # Run kuzu shell one query at a time. This ensures a new process is
-            # created for each query to avoid memory leaks.
-            subprocess.run([kuzu_exec_path, serialized_graph_path],
-                           input=(s + ";" + "\n").encode("ascii"), check=True)
-        except subprocess.CalledProcessError as e:
+        # Run kuzu shell one query at a time. This ensures a new process is
+        # created for each query to avoid memory leaks.
+        process = subprocess.Popen([kuzu_exec_path, serialized_graph_path],
+            stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+        process.stdin.write((s + ";" + "\n").encode("ascii"))
+        process.stdin.close()
+        match = re.match('copy (.*) from', s, re.IGNORECASE)
+        for line in iter(process.stdout.readline, b''):
+            print(line.decode("utf-8"), end='')
```

Comment:
Try to switch to `sys.stdout.write` and ` sys.stdout.flush` to see if it solves the out of order issue